compiler/rustc_mir_transform/src/coverage/tests.rs

```diff
@@ -644,39 +644,15 @@ fn test_traverse_coverage_with_loops() {
     );
 }
 
-fn synthesize_body_span_from_terminators(mir_body: &Body<'_>) -> Span {
-    let mut some_span: Option<Span> = None;
-    for (_, data) in mir_body.basic_blocks.iter_enumerated() {
-        let term_span = data.terminator().source_info.span;
-        if let Some(span) = some_span.as_mut() {
-            *span = span.to(term_span);
-        } else {
-            some_span = Some(term_span)
-        }
-    }
-    some_span.expect("body must have at least one BasicBlock")
-}
-
 #[test]
 fn test_make_bcb_counters() {
     rustc_span::create_default_session_globals_then(|| {
         let mir_body = goto_switchint();
-        let body_span = synthesize_body_span_from_terminators(&mir_body);
         let mut basic_coverage_blocks = graph::CoverageGraph::from_mir(&mir_body);
-        let mut coverage_spans = Vec::new();
-        for (bcb, data) in basic_coverage_blocks.iter_enumerated() {
-            if let Some(span) = spans::filtered_terminator_span(data.terminator(&mir_body)) {
-                coverage_spans.push(spans::CoverageSpan::for_terminator(
-                    spans::function_source_span(span, body_span),
-                    span,
-                    bcb,
-                    data.last_bb(),
-                ));
-            }
-        }
+        let bcb_has_coverage_spans = |bcb: BasicCoverageBlock| bcb.as_usize() > 0;
```

Comment:
This is the part where, instead of trying to make up some `CoverageSpan` instances, we just pretend that every BCB other than 0 has some associated spans.

As far as I can tell, the previous code was calling into `spans`, but it wasn't actually *testing* that code; it was just relying on implementation details. Trying to instead call the span-extraction APIs normally just results in different output, because the synthetic MIR isn't “realistic” enough to give the expected results.

Reply:
Specifically, it would end up producing two coverage spans, one with `bcb = 1` and one with `bcb = 2`.